test/unit/entrypoint/test_main.py

```diff
@@ -13,9 +14,31 @@ def entrypoint_method(name: str) -> str:
 
 def test_no_args(mocker):
     mocker.patch("sys.argv", ["app"])
-    mocker.patch(entrypoint_method("sleep_inifinity"))
+    mocker.patch(entrypoint_method("sleep_infinity"))
     entrypoint.main()
-    assert entrypoint.sleep_inifinity.called
+    assert entrypoint.sleep_infinity.called
+
+
+def test_user_arg(mocker):
+    mocker.patch("sys.argv", [
+        "app",
+        "--user", "jennifer",
+        "--group", "users",
+        "--docker-group", "docker",
+    ])
+    user = create_autospec(entrypoint.User)
+    mocker.patch(entrypoint_method("User"), return_value=user)
+    user.own.return_value = user
+    mocker.patch(entrypoint_method("sleep_infinity"))
+    entrypoint.main()
+    assert entrypoint.User.called
+    args = entrypoint.User.call_args[0]
```

Comment:
You could make group comparable and compare the args as list, is maybe a bit more concise

Reply:
See next push